alacritty_terminal/src/term/mod.rs

```diff
@@ -417,27 +395,25 @@ impl<T> Term<T> {
         }
     }
 
+    /// Collect the information about the changes in the lines, which
+    /// could be used to minimize the amount of drawing operations.
+    ///
+    /// The user controlled elements, like `Vi` mode cursor and `Selection` **are not** part of the
+    /// collected damage state. Those could easily be tracked by comparing their old and new
+    /// value between adjacent frames.
+    ///
+    /// After reading damage [`reset_damage`] should be called.
```

Comment:
Just to allow to call this function more than once, like maybe you want to check that some part is damaged and run custom logic. In our use case it's always _take_ though.

Reply:
This function isn't particularly cheap though, so wouldn't you instead take the damage and then operate on that, instead of calling this function multiple times?

It also mutates the damage based on the pending state, so it's not idempotent.

Reply:
I'm not sure I like the extra complexity with doing the `take` approach, simply because you need to attach the `&mut` to the iterator itself and then reset damage when you drop it, it's just messy.

The way `&mut` is done in this function is because you need current cursor to _finalize_ the damage tracking state, otherwise it'll be wrong. The computation is not that complex though, since it basically does some assignments to make the damage valid.